src/java.base/share/classes/java/lang/foreign/MemorySession.java

```diff
@@ -228,56 +228,85 @@ default MemorySegment allocate(long bytesSize, long bytesAlignment) {
         Objects.requireNonNull(cleaner);
         return MemorySessionImpl.createConfined(Thread.currentThread(), cleaner);
     }
 
     /**
-     * Creates a closeable shared memory session.
-     * @return a new closeable shared memory session.
+     * Creates and returns a new closeable memory session that can be shared across threads.
+     * <p>
+     * The returned memory session can be used by any thread. Users are responsible for
+     * assuring thread-safety across threads that use objects associated with this memory session.
+     * <p>
+     * The returned memory session must eventually be closed to prevent memory leaks.
+     *
+     * @return a new closeable memory session that can be shared across threads
      */
     static MemorySession openShared() {
         return MemorySessionImpl.createShared(null);
     }
 
     /**
-     * Creates a closeable shared memory session, managed by the provided cleaner instance.
-     * @param cleaner the cleaner to be associated with the returned memory session.
-     * @return a new closeable shared memory session, managed by {@code cleaner}.
+     * Creates and returns a new closeable memory session that can be shared across threads,
+     * managed by the provided {@code cleaner}.
+     * <p>
+     * The returned memory session can be used by any thread. Users are responsible for
+     * assuring thread-safety across threads that use objects associated with this memory session.
+     *
+     * @param cleaner the cleaner to be associated with the returned memory session
+     * @return a new closeable memory session that can be shared across threads, managed
+     * by the provided {@code cleaner}
      */
     static MemorySession openShared(Cleaner cleaner) {
         Objects.requireNonNull(cleaner);
         return MemorySessionImpl.createShared(cleaner);
     }
 
     /**
-     * Creates a non-closeable shared memory session, managed by a private {@link Cleaner} instance.
-     * Equivalent to (but likely more efficient than) the following code:
+     * Creates and returns a new non-closeable memory session that can be shared across threads,
+     * manage by a private {@link Cleaner} instance.
+     * <p>
+     * This is equivalent to (but likely more efficient than) the following code:
      * {@snippet lang=java :
      * openShared(Cleaner.create()).asNonCloseable();
      * }
-     * @return a non-closeable shared memory session, managed by a private {@link Cleaner} instance.
+     * @return a non-closeable, thread-shared memory session, managed by a private {@link Cleaner} instance
      */
     static MemorySession openImplicit() {
         return MemorySessionImpl.createImplicit();
     }
 
     /**
      * Returns the global memory session.
-     * @return the global memory session.
+     * <p>
+     * The global memory session is non-closeable and can be shared across threads.
+     * <p>
+     * The global memory session is neither ever closed nor ever subject to cleaning.
```

Comment:
These comments seem out of place, in the sense that the concept of memory session is defined in the top level javadoc. I'm not sure repeating what a global session is here adds much? If we really think that the old text is not good enough, then maybe pulling out the global session text into a subsection in the toplevel javadoc, and then referring to that one from here could be a better option (although, again, not sure this needs changing?).